### PR TITLE
Adding system diagnostics process tests

### PR DESCRIPTION
**Pull request by MattGal**
Initial commit of System.Diagnostics.Process tests, as well as a fix in System.Diagnostics.Process for single-core machines

### CODE REVIEW

**Comment by MattGal** [comment]
comment to see if Jenkin's will retry tests...

**Comment by MattGal** [comment]
test this please

**Comment by mmitche** [comment]
test this please

**Comment by joshfree** [comment]
1. please submit initial commits as dotnet-bot for git blame purposes
2. please flatten the commit history and have the commit message follow the guidelines here https://github.com/dotnet/corefx/wiki/Contributing 

```
Summarize change in 50 characters or less

Provide more detail after the first line. Leave one blank line below the
summary and wrap all lines at 72 characters or less.

If the change fixes an issue, leave another blank line after the final
paragraph and indicate which issue is fixed in the specific format
below.

Fix #42
```


For this I'd prefer the message simply be

```
Initial Commit of System.Diagnostic.Process tests
```

:smile:

File: src/System.Diagnostics.Process.sln [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 2013
-VisualStudioVersion = 12.0.30723.0
+# Visual Studio 14
+VisualStudioVersion = 14.0.22512.0
```
Comment (joshfree) [inline src/System.Diagnostics.Process.sln]: All of our other solutions are VS 2013; please keep this on solution format version 12.00

File: src/System.Diagnostics.Process/tests/ProcessTest_ConsoleApp/ProcessTest_ConsoleApp.csproj [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <ProjectGuid>{69E46A6F-9966-45A5-8945-2559FE337827}</ProjectGuid>
+    <OutputType>Exe</OutputType>
+    <RootNamespace>ProcessTest_ConsoleApp</RootNamespace>
+    <AssemblyName>ProcessTest_ConsoleApp</AssemblyName>
+    <TargetFrameworkVersion>4.5</TargetFrameworkVersion>
+    <OutputPath Condition="'$(OutputPath)'==''">..\..\..\..\bin\$(Configuration)\System.Diagnostics.Process.Tests\</OutputPath>
```
Comment (joshfree) [inline src/System.Diagnostics.Process/tests/ProcessTest_ConsoleApp/ProcessTest_ConsoleApp.csproj]: can we do something better than the 
```
..\..\..\..\bin\
```

**Comment by joshfree** [comment]
Can you also make the tests not Console.WriteLine on success/expected cases (only on failures).  I know the Regex tests don't currently follow this either - but those are bugged and will be fixed.  

Specifically these 5 Console.WriteLine's
```
15:43:22   ProcessTest_ConsoleApp.exe started
15:43:22   ProcessTest_ConsoleApp.exe closed
15:43:22   ProcessTest_ConsoleApp.exe started
15:43:22   ProcessTest_ConsoleApp.exe closed
15:43:22   .
```

**Comment by joshfree** [comment]
:+1 assuming issues above are addressed.

**Comment by MattGal** [comment]
The Console.WriteLine()s there are needed to test variations of Standard output behaviors... redirecting it or not having it would change the nature of the test.  Other issues addressed.

**Comment by joshfree** [comment]
Awesome thanks Matt

**Pull request merged**